src/node.cc

```diff
@@ -2548,6 +2548,7 @@ Isolate* NewIsolate(ArrayBufferAllocator* allocator, uv_loop_t* event_loop) {
   isolate->SetMicrotasksPolicy(MicrotasksPolicy::kExplicit);
   isolate->SetFatalErrorHandler(OnFatalError);
   isolate->SetAllowWasmCodeGenerationCallback(AllowWasmCodeGenerationCallback);
+  v8::CpuProfiler::UseDetailedSourcePositionsForProfiling(isolate);
```

Comment:
Question: does this have a penalty? And even if not could this be conditional on inspector being attached or profiler run requested?

Reply:
The problem with only enabling this once the profiler run is requested is that functions that get optimized before profiling starts don't have the detailed line information. In the browser you can work around this by reloading the page with profiling enabled (devtools has a button for this) but for a running Node instance this is not really feasible, particularly if you want to debug some rarely-occurring thing. Unfortunately we see this in the wild quite a bit so having it on by default makes sense IMO.

The overhead is very low - on average it makes no difference, it the worst case a few percent increase in optimized compile time, but this is currently being moved off-thread anyway (plus the general performance boost of upgrading the V8 version is going to easily outweigh this).

Reply:
Thanks for the explanation.

Reply:
CpuProfiler can be also used without inspector, see e.g. NPM modules like [v8-profiler](https://www.npmjs.com/package/v8-profiler-node8). There are tools out in the wild which start profiling automatically for some time if they notice performance issues.

Reply:
It's also possible to use the CPU profiler directly through the V8 API without the inspector and there are some native modules that do this - that approach wouldn't work for them as far as I understand

Reply:
The thoughts leading to this were:
- It's hard to foresee whether the user is going to profile, both for Chrome and Node.
- For Chrome, you can at least reload, with DevTools open.
- Node.js does not have the reload option.
- Adding more accurate data has a small one-time cost for the script. Chrome suffers a bit more for this as you pay more for optimizing compiles, possibly while starting up a page, and this occurs on every page navigation. Node.js does not encounter new scripts to compile and optimize as often.

Reply:
I agree that it's a tough call:
slightly slower compile time but for everyone ⚖️ improvement not available but only for profiling started out-of-band